logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/TypingIndicatorRepository.kt

```diff
@@ -0,0 +1,75 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.kalium.logic.data.conversation
+
+import co.touchlab.stately.collections.ConcurrentMutableMap
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.user.UserId
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.distinctUntilChanged
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.onStart
+import kotlinx.datetime.Clock
+import kotlinx.datetime.Instant
+
+interface TypingIndicatorRepository {
+    fun addTypingUserInConversation(conversationId: ConversationId, userId: UserId)
+    fun removeTypingUserInConversation(conversationId: ConversationId, userId: UserId)
+    suspend fun observeUsersTyping(conversationId: ConversationId): Flow<Set<UserId>>
+}
+
+class TypingIndicatorRepositoryImpl : TypingIndicatorRepository {
+
+    private val userTypingDataSourceFlow = MutableSharedFlow<Unit>(extraBufferCapacity = 1, onBufferOverflow = BufferOverflow.DROP_OLDEST)
+    private val userTypingCache = ConcurrentMutableMap<ConversationId, Set<ExpiringUserTyping>>()
```

Comment:
This makes it so this Repository needs to be "static" _a.k.a._ we need to create one for each user, can't just have getters like most repositories.

It would be nice to make this more explicit by passing the `MutableMap<ConversationId, Set<ExpiringUserTyping>>` as a parameter in the constructor.

Reply:
Done !